kmod/src/forest.c

```diff
@@ -348,15 +348,9 @@ int scoutfs_forest_read_items(struct super_block *sb,
 	ret = 0;
 out:
 	if (ret == -ESTALE) {
-		if (memcmp(&prev_refs, &refs, sizeof(refs)) == 0) {
-			ret = -EIO;
-			goto out;
-		}
+		if (memcmp(&prev_refs, &refs, sizeof(refs)) == 0)
+			return -EIO;
```

Comment:
Yeah, making it match the other retry handling makes sense, though I don't really like constructions with multiple returns when we can avoid it.